File: app/src/main/java/com/rowland/delivery/features/dash/domain/models/category/Category.java
package com.rowland.delivery.features.dash.domain.models.category;

import android.support.annotation.NonNull;

import com.google.firebase.firestore.Exclude;
import com.rowland.delivery.features.dash.domain.models.FirestoreModel;

/**
 * Created by Rowland on 5/5/2018.
 */

public class Category implements FirestoreModel {

    private String name;

    @Exclude
    private String firestoreUid;

    @Override
    public <T extends FirestoreModel> T withFirestoreId(@NonNull String firestoreUid) {
        this.firestoreUid = firestoreUid;
        return (T) this;
    }

    public String getFirestoreUid() {
        return firestoreUid;
    }

    public Category() {
    }

    public Category(String name) {
        this.name = name;
    }

    public String getName() {
        return name;
    }

    public void setName(String name) {
        this.name = name;
    }

    @Override
    public String toString() {
        return "CategoryScope{" +
                "name='" + name + '\'' +
                '}';
    }
}

File: app/src/main/java/com/rowland/delivery/features/dash/domain/models/category/Category.kt
package com.rowland.delivery.features.dash.domain.models.category

import com.google.firebase.firestore.Exclude
import com.rowland.delivery.features.dash.domain.models.FirestoreModel

/**
 * Created by Rowland on 5/5/2018.
 */

class Category : FirestoreModel {

    var name: String? = null

    @Exclude
    var firestoreUid: String? = null
        private set

    override fun <T : FirestoreModel> withFirestoreId(firestoreUid: String): T {
        this.firestoreUid = firestoreUid
        return this as T
    }

    constructor()

    constructor(name: String) {
        this.name = name
    }

    override fun toString(): String {
        return "CategoryScope{" +
                "name='" + name + '\''.toString() +
                '}'.toString()
    }
}
